joomla/joomla-cms
# [5.1] Fix duplicate Brotli .htaccess postinstall message after update from 4.4

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) fixes the duplicate postinstall message for the Brotli .htaccess message which you get after updating from 4.4.3 or the latest 4.4 nightly build to the latest 5.1-dev nightly build.

`INSERT IGNORE` (MySQL and MariaDB) or `ON CONFLICT DO NOTHING` (PostgreSQL) don't help here as we do not use the `postinstall_message_id` column in the inserts, and so no primary key or unique key would be violated when inserting multiple times.

This PR fixes that by adding a condition for that to the `SELECT` part of the `INSERT INTO .. SELECT` statement.

For the reason mentioned above, it also removes the `IGNORE` from the MySQL and the `ON CONFLICT DO NOTHING` from the PostgreSQL script.

The duplicate happens when updating from 4.4.3 or any older 4.4.x because the 5.1.0 nightly package includes the `4.4.4-2024-03-28.sql` update SQL script from the latest upmerge, so that one and the one modified by this PR are run both.

Depending on if we require to be on 4.4.4 for updating to 5.1 or if we allow updating to 5.1 also for older 4.4.x versions, we should remove the `4.4.4-2024-03-28.sql` update SQL script from the 5.1-dev branch and delete it on update, or we should not do that and keep it.

Currently the target platform values of the 5.1.0 Beta targets in https://update.joomla.org/cms/targets.json only require to be on 4.4.x, they do not require 4.4.4. If we would require that, we would have to update the targetplatform of the 5.x versions with every new 4.4. version, and in future do the same with 6.x for every new 5.y.

So maybe we should not require the latest version and keep the `4.4.4-2024-03-28.sql` script in 5.x-dev branches.

I will clarify that with other maintainer and release managers.

@brianteeman Seems you were right with your worries.

### Testing Instructions

Update from 4.4.3 or any older 4.4.x to the latest 5.1 nightly build for the actual result (you have to use "Upload & Update"), and update from 4.4.3 or any older 4.4.x to the patched update package or custom update URL created by Drone for this PR for the expected result.

The latest 5.1 nightly build can be downloaded from here: https://developer.joomla.org/nightlies/Joomla_5.1.0-beta3-dev-Development-Update_Package.zip

The patched update package or custom update URL created by Drone for this PR can be found here: https://artifacts.joomla.org/drone/joomla/joomla-cms/5.1-dev/43182/downloads/75140/

### Actual result BEFORE applying this Pull Request

![2024-03-30_j5 1-duplicate-postinstall-msg](https://github.com/joomla/joomla-cms/assets/7413183/9f982a36-474f-4d5b-b787-38c8afcbadf2)

### Expected result AFTER applying this Pull Request

Only one message.

### Link to documentations
Please select:
